### ISSUE TITLE
Batch read resource when refreshing

### ISSUE THREAD

**Issue by chenzejun** [opentofu/opentofu#1278]
### OpenTofu Version

```shell
1.6.1
```

### Use Cases

I have a big stack with a large number of resources
Everytime when I want to do/plan some change  by opentofu, opentofu will read all the resources separately. 
This will bring a lot of read calls to the cloud server, cause long time to run even if there is a small change.

### Attempted Solutions

no attempted solutions

### Proposal

Add the batch read ability to the providers.
```
    rpc ReadResources(ReadResources.Request) returns (ReadResources.Response);
```
Add the max batch number to the provider sechme to tell opentofu this resource type supports batch read and how many resources can be read in one rpc/api call
Opentofu can call ReadResources rpc to refresh resources if this resource type supports batch read.

**Comment by chenzejun** [comment]
Btw, I can help contribute to this feature if this is accepted :)

**Comment by janosdebugs** [comment]
Hi @chenzejun it is unlikely we can add this feature because it would be a change that's incompatible with Terraform and provider authors would have to decide one or the other. Currently, we (the core team) get a lot of feedback that compatibility is very important for now, which means we can't implement this. Maybe in the future.

**Comment by chenzejun** [comment]
@janosdebugs Thanks for reply!
As an enhancement feature. I think we can make it optional for opentofu to make it backward compatible.
Opentofu can work with both preexisting provider(based on Terraform framework) and batch-read provider(base on the opentofu framework). 
If any provider developer needs this feature, can choose to use opentofu framework to imporve the performance on openfotu.
And the provider developers who are not interest in this ferture, need not make any choice.

**Comment by janosdebugs** [comment]
@chenzejun that would mean forking the provider framework and SDK, which would need steering committee approval (see "Can you fork project X?" in the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md#can-you-fork-project-x)).

**Comment by chenzejun** [comment]
Thanks @janosdebugs! And if it's a development capacity concern, I can help do this.

**Comment by janosdebugs** [comment]
Hey @chenzejun we discussed this in the core team and unfortunately, we decided unanimously that we will not take up this issue at this time. This feature would require us to fork several libraries from HashiCorp, which is a long-term commitment for the core team which cannot be outsourced. Additionally, this would require wide-ranging changes to the graph algorithm in OpenTofu, which we cannot dedicate enough capacity to plan, review, document, and generally support the development.
